clock-impl/src/main/java/io/deephaven/clock/impl/JdkInternalsRealTimeClock.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.clock.impl;
+
+import com.google.auto.service.AutoService;
+import io.deephaven.clock.RealTimeClock;
+import jdk.internal.misc.VM;
+
+import java.time.Instant;
+
+/**
+ * Clock implementation derived from {@link java.time.Clock#systemUTC()}. The {@link #currentTimeNanos()} method returns
+ * the same time represented by {@link Instant#now()} without allocation.
+ */
+
+@AutoService(RealTimeClock.class)
+public final class JdkInternalsRealTimeClock implements RealTimeClock {
+
+    @Override
+    public long currentTimeMicros() {
+        return currentTimeNanos() / 1000L;
+    }
+
+    // We don't actually need a volatile here.
+    // We don't care if offset is set or read concurrently by multiple
+    // threads - we just need a value which is 'recent enough' - in other
+    // words something that has been updated at least once in the last
+    // 2^32 secs (~136 years). And even if we by chance see an invalid
+    // offset, the worst that can happen is that we will get a -1 value
+    // from getNanoTimeAdjustment, forcing us to update the offset
+    // once again.
+    private transient long offset;
+
+    /**
+     * Returns the current time in nanoseconds. This is the same moment in time {@link Instant#now()} would return, but
+     * represented (without allocation) as a single {@code long} instead of as an {@link Instant} object.
+     * 
+     * @return the difference, measured in nanoseconds (with precision determined by the result of
+     *         {@link VM#getNanoTimeAdjustment}), between the current time and midnight, January 1, 1970 UTC.
+     */
+    @Override
+    public long currentTimeNanos() {
```

Comment:
We should test out some different strategies. We _may_ be able to get away w/ simply:
```java
return VM.getNanoTimeAdjustment(0);
```